S2VX.Game/TimelineBar.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using JetBrains.Annotations;
+using osu.Framework.Allocation;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Input.Events;
+using osuTK.Graphics;
+
+namespace S2VX.Game
+{
+    public class TimelineBar : Box
+    {
+        [Resolved]
+        private Story Story { get; set; }
```

Comment:
Try and always set declarations = to something

Maybe prefer
`private Story Story { get; set; } = new Story()`

Ideally, it's best practice to never initialize something as null, but sometimes in the case of asynchronous loading this is harder to avoid

I don't think the above will work because Story may reference itself, in that case we can set this to null
`private Story Story { get; set; } = null`